### DISCUSSION TITLE
Let's release 3.3.2

### DISCUSSION THREAD

**Discussion by objecttothis** [opensourcepos/opensourcepos#3676] [Team Posts]
Unless I'm missing something, we have PR #2840 (@WebShells) and maybe #2780 (@SteveIreland) needed to get merged for 3.3.2 to get released.  Are there other things that I'm missing?  #2840 just has two lines of code that need to get changed and #2780 may or may not need to get merged for 3.3.2... it depends on whether its fixing a bug or providing an enhancement.

**Comment by objecttothis** [comment]
It looks like issues #2819 and #2820 are labeled for 3.3.2 and not currently addressed by a pull request.

**Comment by objecttothis** [comment]
I went ahead and made the last changes on #2840 then merged it.  @SteveIreland, what's the status on #2780? @jekkos, what's the status of #2819 and #2820?  Do you want me to take #2819?

**Comment by jekkos** [comment]
I'll have a look at the invoice number thingy, wanted to do that today but didn't get to it. But feel free to check.

**Comment by objecttothis** [comment]
@jekkos I looked some at #2819 today and got it narrowed down.  It has something to do with the way that you are querying the attribute values.  I wrote a few comments on that issue.

**Comment by SteveIreland** [comment]
I'm going to review/test/update/merge #2780 today ... I know @jekkos wanted me to incorporate a PHP Unit test (and so did I)  but work/life have prevented me from being able to make time for it.  So unless someone stops me that is my plan for today.

**Comment by objecttothis** [comment]
> I'm going to review/test/update/merge #2780 today ... I know @jekkos wanted me to incorporate a PHP Unit test (and so did I) but work/life have prevented me from being able to make time for it. So unless someone stops me that is my plan for today.

Sounds good to me.  See my note on the PR.  You mention that you made the one fix I pointed out, but perhaps it still needs to get pushed to the branch on the repository.

**Comment by objecttothis** [comment]
OK, #2819 is going to get a temporary patch (waiting for a code review from anyone who wants to sign off on one line addition) but is essentially going to get punted until 3.4.0 for a full fix.  @SteveIreland appears to have everything in the bag with #2780.  That just leaves #2820 which I think @jekkos is handling and we can release 3.3.2.

**Comment by SteveIreland** [comment]
Yes, I'm still testing and debugging.

**Comment by jekkos** [comment]
Ok I think we then have two things open now? I did not get any feedback on the tax issues so presume all is fine there? Also I'll try to close that last PR #2860 this week so we can finally ship the release?

**Comment by jekkos** [comment]
Funny I was looking in docker hub and saw we already have a 3.3.2 tag in there. Probably that is because we tend to bump to the next version and then this one is marked a 'final'. Maybe we should try to adhere to a SNAPSHOT or -dev suffix while we develop on master branch so that the created builds can be clearly distinguished? I tihnk I have added this suffix alredy in the footer if it's not an official release so maybe use that approach in git as well?

**Comment by objecttothis** [comment]
@jekkos can you clarify?  Do you mean that at the beginning of the dev cycle we create a branch called master-dev or master-SNAPSHOT and only merge into that?

**Comment by jekkos** [comment]
well I think we should have done it this way, but it might be hard to correct history as I'd like to have this aligned.
I was checking whether I can fetch a release tag from docker hub and retrieve the exact version for that relaese.

curerntly the last version tag I find is 3.3.1 (pushed 7 months ago). So it does look Ok. If we move to 3.4 we should have 3.3.2 tagged there that can then be deployed on the dev server.

docker pull jekkos/opensourcepos:3.3.1
